src/LaserTag.jl

```diff
@@ -143,6 +143,11 @@ function POMDPs.reward(p::LaserTagPOMDP, s::LTState, a::Int, sp::LTState)
     end
 end
 
+# add a function to transform a LTState to a vector 
+function POMDPs.convert_s(T::Type{<:AbstractArray}, s::LTState, p::LaserTagPOMDP)
+    return convert(T, vcat(s.robot, s.opponent, [s.terminal]))
```

Comment:
```suggestion
    return convert(T, vcat(s.robot, s.opponent, s.terminal))
```
I don't think you need to create a vector with s.terminal in it. That will cause an additional allocation.

(additionally, I think `vcat` itself might cause an allocation, which is disappointing. You *might* be able to eliminate all allocations with StaticArrays, but that might take a bit of learning. There are many tools to track allocations including `BenchmarkTools.@btime`

Reply:
Thanks for the info!